geopandas/tools/clip.py

```diff
@@ -37,41 +50,94 @@ def _clip_points(gdf, poly):
         The returned GeoDataFrame is a subset of gdf that intersects
         with poly.
     """
-    return gdf.iloc[gdf.sindex.query(poly, predicate="intersects")]
+    clipping_candidates = gdf.iloc[gdf.sindex.query(poly, predicate="intersects")]
+    point_indices = clipping_candidates.geom_type == "Point"
+    # For performance reasons points don't need to be intersected with poly
+    points = clipping_candidates[point_indices]
 
+    non_points = clipping_candidates[~point_indices]
+    non_points_intersected = _intersect_non_point_geometries_with_poly(non_points, poly)
+    return pd.concat([points, non_points_intersected])
 
-def _clip_line_poly(gdf, poly):
-    """Clip line and polygon geometry to the polygon extent.
 
-    Clip an input line or polygon to the polygon extent of the poly
-    parameter. Parts of Lines or Polygons that intersect the poly geometry are
-    extracted with associated attributes and returned.
+def _validate_inputs(gdf, mask):
+    if not isinstance(gdf, (GeoDataFrame, GeoSeries)):
+        raise TypeError(
+            "'gdf' should be GeoDataFrame or GeoSeries, got {}".format(type(gdf))
+        )
 
-    Parameters
-    ----------
-    gdf : GeoDataFrame, GeoSeries
-        Line or polygon geometry that is clipped to poly.
+    if not isinstance(mask, (GeoDataFrame, GeoSeries, Polygon, MultiPolygon)):
+        raise TypeError(
+            "'mask' should be GeoDataFrame, GeoSeries or"
+            "(Multi)Polygon, got {}".format(type(mask))
+        )
+    if isinstance(mask, (GeoDataFrame, GeoSeries)):
+        if not _check_crs(gdf, mask):
+            _crs_mismatch_warn(gdf, mask, stacklevel=3)
```

Comment:
```suggestion
            _crs_mismatch_warn(gdf, mask, stacklevel=4)
```

We're one level deeper now.